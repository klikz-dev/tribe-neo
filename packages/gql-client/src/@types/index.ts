import { ClientError as CError } from 'graphql-request'

export enum ApiErrorCodes {
  UNKNOWN_ERROR = 1,
  SERVER_ERROR = 10,
  PARAMETER_VALIDATION_FAILED = 100,
  PERMISSION_ERROR = 102,
  USER_REQUEST_REACHED = 104,
  INVALID_PARAMETER = 106,
  INVALID_ACCESS_TOKEN = 108,
  RESOURCE_NOT_FOUND = 110,
  MISSING_PARAMETER = 112,
  PARAMETER_TYPE_PROBLEM = 114,
  DATABASE_ERROR = 116,
  INVALID_REQUEST = 118,
  REQUEST_BLOCKED_FOR_POLICIES = 120,
  BACKEND_ERROR = 122,
}

export type ErrorResponse = {
  code: ApiErrorCodes
  field?: string
  message: string
  subcode?: string
  help?: string
  timestamp?: string
}

interface ClientResponse<T = any> {
  data?: T
  errors?: ErrorResponse[]
  extensions?: any
  status: number
  [key: string]: any
}

export type ClientError = Omit<CError, 'response'> & {
  response: ClientResponse
}

export type ClientAttachmentError = {
  fileName: string
  message?: string
}

export type ClientAttachmentErrorHandler = (
  error: ClientAttachmentError,
) => void
