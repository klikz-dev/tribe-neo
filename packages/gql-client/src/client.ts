import {
  AppClient,
  AuthClient,
  BillingClient,
  CollectionsClient,
  DevClient,
  GraphqlClient,
  InvitationsClient,
  MediaClient,
  MembersClient,
  ModerationClient,
  NetworkClient,
  NotificationsClient,
  PagesClient,
  PostsClient,
  ReportClient,
  RolesClient,
  SlatesClient,
  SpaceMembersClient,
  SpaceMembershipClient,
  SpacePostTypeClient,
  SpaceRolesClient,
  SpacesClient,
  TagsClient,
  ThemeClient,
  TribeClientOptions,
} from './clients'
import {
  AuthTokenFields,
  getLimitedTokenGQLQuery,
  getTokensGQLQuery,
  LimitedTokenFields,
  SearchEntityFields,
  searchGQLQuery,
} from './graphql'
import {
  AppToken,
  AuthToken,
  PermissionContext,
  QueryLimitedTokenArgs,
  QuerySearchArgs,
  QueryTokensArgs,
  SearchResult,
} from './types'

export class TribeClient {
  app: AppClient

  auth: AuthClient

  billing: BillingClient

  collections: CollectionsClient

  dev: DevClient

  invitations: InvitationsClient

  media: MediaClient

  members: MembersClient

  moderation: ModerationClient

  network: NetworkClient

  notifications: NotificationsClient

  posts: PostsClient

  report: ReportClient

  roles: RolesClient

  spaceMembers: SpaceMembersClient

  spaceMembership: SpaceMembershipClient

  spaceRoles: SpaceRolesClient

  spaces: SpacesClient

  spacePostType: SpacePostTypeClient

  tags: TagsClient

  theme: ThemeClient

  slates: SlatesClient

  pages: PagesClient

  private client: GraphqlClient

  constructor(options: TribeClientOptions) {
    this.client = new GraphqlClient(options)

    this.app = new AppClient(this.client)
    this.auth = new AuthClient(this.client)
    this.billing = new BillingClient(this.client)
    this.collections = new CollectionsClient(this.client)
    this.dev = new DevClient(this.client)
    this.invitations = new InvitationsClient(this.client)
    this.media = new MediaClient(this.client)
    this.members = new MembersClient(this.client)
    this.moderation = new ModerationClient(this.client)
    this.network = new NetworkClient(this.client)
    this.notifications = new NotificationsClient(this.client)
    this.posts = new PostsClient(this.client)
    this.report = new ReportClient(this.client)
    this.roles = new RolesClient(this.client)
    this.spaceMembers = new SpaceMembersClient(this.client)
    this.spaceMembership = new SpaceMembershipClient(this.client)
    this.spaceRoles = new SpaceRolesClient(this.client)
    this.spaces = new SpacesClient(this.client)
    this.tags = new TagsClient(this.client)
    this.theme = new ThemeClient(this.client)
    this.slates = new SlatesClient(this.client)
    this.pages = new PagesClient(this.client)
    this.spacePostType = new SpacePostTypeClient(this.client)
  }

  setToken(accessToken: string): void {
    this.client.setToken(accessToken)
  }

  /**
   * Get the guest access token for a community.
   * @query tokens(): AuthToken!
   * @example
   * ```typescript
   *  t.client.getTokens({networkDomain: 'community.tribe.so'})
   * ```
   * @returns The AuthToken.
   */
  async getTokens(
    variables: QueryTokensArgs,
    fields: AuthTokenFields = 'default',
  ): Promise<AuthToken> {
    type QueryResult = { tokens: AuthToken }
    const result = await this.client.authorizedRequest<QueryResult>({
      query: getTokensGQLQuery(fields),
      variables,
    })
    return result.tokens
  }

  async getLimitedToken(
    variables: QueryLimitedTokenArgs,
    fields: LimitedTokenFields = 'basic',
    useBasicToken?: boolean,
  ): Promise<AppToken> {
    type QueryResult = { limitedToken: AppToken }
    const result = await this.client.authorizedRequest<QueryResult>({
      query: getLimitedTokenGQLQuery(fields),
      variables,
      useBasicToken,
    })
    return result.limitedToken
  }

  async search(
    variables: QuerySearchArgs,
    fields: SearchEntityFields = 'basic',
    accessToken?: string,
  ): Promise<SearchResult> {
    type QueryResult = { search: SearchResult }
    const result = await this.client.authorizedRequest<QueryResult>({
      query: searchGQLQuery(fields),
      variables,
      customToken: accessToken,
    })
    return result.search
  }

  async generateToken(options: {
    networkId: string
    memberId?: string
  }): Promise<string> {
    const { networkId, memberId } = options
    return this.getLimitedToken(
      {
        context: PermissionContext.NETWORK,
        networkId,
        entityId: networkId,
        impersonateMemberId: memberId,
      },
      'basic',
      true,
    ).then(x => x.accessToken)
  }
}
