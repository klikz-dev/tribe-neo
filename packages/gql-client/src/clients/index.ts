export * from './app.client'
export * from './auth.client'
export * from './billing.client'
export * from './collections.client'
export * from './dev.client'
export * from './graphql.client'
export * from './invitations.client'
export * from './media.client'
export * from './members.client'
export * from './moderation.client'
export * from './network.client'
export * from './notifications.client'
export * from './pages.client'
export * from './posts.client'
export * from './report.client'
export * from './roles.client'
export * from './slates.client'
export * from './space-post-type.client'
export * from './spaceMembers.client'
export * from './spaceMembership.client'
export * from './spaceRoles.client'
export * from './spaces.client'
export * from './tags.client'
export * from './theme.client'
