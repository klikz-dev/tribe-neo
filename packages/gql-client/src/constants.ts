export const BASE_URL =
  typeof process !== 'undefined' ? process?.env?.TRIBE_GQL_ENDPOINT : ''
