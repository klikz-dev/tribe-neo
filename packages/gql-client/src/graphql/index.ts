export * from './app'
export * from './auth'
export * from './billing'
export * from './dev'
export * from './media'
export * from './moderation'
export * from './network'
export * from './notification'
export * from './page'
export * from './permission'
export * from './post'
export * from './report'
export * from './search'
export * from './slate'
export * from './space'
export * from './theme'
