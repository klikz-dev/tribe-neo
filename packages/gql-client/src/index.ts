export * from './@types/index'
export * from './client'
export * from './graphql'
export * as Types from './types'
export * from './utils'
