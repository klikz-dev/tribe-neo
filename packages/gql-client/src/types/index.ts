import { CreateFileInput, CreateImageInput } from './tribe-graphql.generated'

export * from './tribe-graphql.generated'
export * from './fields'

export type UploadImagesArgs = Omit<CreateImageInput, 'contentType'> & {
  contentType?: string
  file: File
}

export type AttachmentUploadType = Array<
  CreateFileInput & {
    file: File
  }
>
