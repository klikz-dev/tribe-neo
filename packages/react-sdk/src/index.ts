export * from './@types'
export * from './Provider'
export * from './lib'
export * from './ClientProvider'
export * from './useTribeClient'
export * from './Hydrate'
