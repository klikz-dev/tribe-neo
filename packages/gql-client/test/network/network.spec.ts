import { networkGQLFields } from 'src'

describe('Network GraphQL Fields', () => {
  it('should return correct fields', async () => {
    expect(networkGQLFields('basic')).toMatchSnapshot()
    expect(networkGQLFields('all')).toMatchSnapshot()
    expect(
      networkGQLFields({
        customCode: 'all',
        roles: 'basic',
        defaultSpaces: {
          spaceType: {
            availablePostTypes: {
              validReplyTypes: {
                validReplyTypes: {
                  validReplyTypes: 'basic',
                },
              },
            },
          },
        },
      }),
    ).toMatchSnapshot()
  })
})
