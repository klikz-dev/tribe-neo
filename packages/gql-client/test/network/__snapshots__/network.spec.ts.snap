// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Network GraphQL Fields should return correct fields 1`] = `
"
    id
    companyName
    name
    description
    visibility
    membership
    logoId
    faviconId
    locale
    domain
    newDomain
    incidentEmails
    privacyPolicyUrl
    termsOfServiceUrl
    brandColor
    billingEmail
    status
    memberCapacity
    seatsCapacity
    additionalSeatsCapacity
    seatCapacityDeclared
    tribeBranding
    whoCanInviteIds
    
    
    
    
    
    
    
    
    
  "
`;

exports[`Network GraphQL Fields should return correct fields 2`] = `
"
    id
    companyName
    name
    description
    visibility
    membership
    logoId
    faviconId
    locale
    domain
    newDomain
    incidentEmails
    privacyPolicyUrl
    termsOfServiceUrl
    brandColor
    billingEmail
    status
    memberCapacity
    seatsCapacity
    additionalSeatsCapacity
    seatCapacityDeclared
    tribeBranding
    whoCanInviteIds
    
      themes {
        
    active {
      name
      status
      tokens {
        colors {
          key
          value
        }
      }
    }
  
      }
    
    
      topNavigation {
        
    enabled
    items {
      link
      openInNewWindow
      text
      type
    }
  
      }
    
    
      logo {
        
      ... on Image {
        
    __typename
    id
    url
    width
    height
    dominantColorHex
    dpi
    cropHeight
    cropWidth
    cropX
    cropY
    cropZoom
    urls {
      full
      large
      medium
      small
      thumb
    }
  
      }
      ... on Emoji {
        
    __typename
    id
    text
  
      }
  
      }
    
    
      favicon {
        
      ... on Image {
        
    __typename
    id
    url
    width
    height
    dominantColorHex
    dpi
    cropHeight
    cropWidth
    cropX
    cropY
    cropZoom
    urls {
      full
      large
      medium
      small
      thumb
    }
  
      }
      ... on Emoji {
        
    __typename
    id
    text
  
      }
  
      }
    
    
      whoCanInvite {
        
    id
    name
    type
    description
    visible
    scopes
  
      }
    
    
      defaultSpaces {
        
    id
    name
    description
    slug
    spaceTypeId
    createdById
    groupId
    imageId
    bannerId
    membersCount
    createdAt
    private
    hidden
    inviteOnly
    nonAdminsCanInvite
    customOrderingIndexInGroup
    
    
        
        
        
    
       
     
  
      }
    
    
      roles {
        
    id
    name
    type
    description
    visible
    scopes
  
      }
    
    
      customCodes(anonymize: false) {
        
    anonymize
    position
    code
  
      }
    
    
      authMemberProps {
        
    context
    scopes
    
  
      }
    
  "
`;

exports[`Network GraphQL Fields should return correct fields 3`] = `
"
    id
    companyName
    name
    description
    visibility
    membership
    logoId
    faviconId
    locale
    domain
    newDomain
    incidentEmails
    privacyPolicyUrl
    termsOfServiceUrl
    brandColor
    billingEmail
    status
    memberCapacity
    seatsCapacity
    additionalSeatsCapacity
    seatCapacityDeclared
    tribeBranding
    whoCanInviteIds
    
    
    
    
    
    
      defaultSpaces {
        
    id
    name
    description
    slug
    spaceTypeId
    createdById
    groupId
    imageId
    bannerId
    membersCount
    createdAt
    private
    hidden
    inviteOnly
    nonAdminsCanInvite
    customOrderingIndexInGroup
    
    
    
      spaceType {
        
    id
    name
    
      availablePostTypes {
        
    context
    id
    name
    pluralName
    postTypeTemplateId
    createdAt
    primaryReactionType
    singleChoiceReactions
    updatedAt
    mappings {
      key
      field
      type
      title
      description
      required
      isMainContent
      isSearchable
      default
    }
    validReplyTypesIds
    
      validReplyTypes {
        
    context
    id
    name
    pluralName
    postTypeTemplateId
    createdAt
    primaryReactionType
    singleChoiceReactions
    updatedAt
    mappings {
      key
      field
      type
      title
      description
      required
      isMainContent
      isSearchable
      default
    }
    validReplyTypesIds
    
      validReplyTypes {
        
    context
    id
    name
    pluralName
    postTypeTemplateId
    createdAt
    primaryReactionType
    singleChoiceReactions
    updatedAt
    mappings {
      key
      field
      type
      title
      description
      required
      isMainContent
      isSearchable
      default
    }
    validReplyTypesIds
    
      validReplyTypes {
        
    context
    id
    name
    pluralName
    postTypeTemplateId
    createdAt
    primaryReactionType
    singleChoiceReactions
    updatedAt
    mappings {
      key
      field
      type
      title
      description
      required
      isMainContent
      isSearchable
      default
    }
    validReplyTypesIds
    
  
      }
    
  
      }
    
  
      }
    
  
      }
    
  
      }
        
        
        
    
       
     
  
      }
    
    
      roles {
        
    id
    name
    type
    description
    visible
    scopes
  
      }
    
    
      customCodes(anonymize: false) {
        
    anonymize
    position
    code
  
      }
    
    
  "
`;
